website/docs/utilities/dialog-primitive/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,53 @@
+## Usage
+
+The DialogPrimitive sub-components are not intended to be used independently of each other or in isolation. If you are looking to create your own dialog-based component, use the `DialogPrimitive::Wrapper` as your foundation.
+
+Some examples may include:
+
+- A dialog element that is inline with the main page content and does not overlay the page, often referred to as a "non-modal" dialog.
+- A custom dialog element if neither the [Modal](/components/modal) or [Flyout](/components/flyout) are suitable.
```

Comment:
This may encourage people too much? Maybe add something about contacting us in this case, to make sure the existing components don't cover already their needs?

Reply:
I've cleaned this up and recommended to reach out with use cases outside of this example. https://github.com/hashicorp/design-system/pull/1945/commits/8f81e9890e4087ea20d9d30b44759c7a04776a2f